Implementation/1 implement converter service

Havryliuk-Vasyl opened the pull request.

rdovgan committed:
Added test class for IsConverterService

rdovgan committed:
Merge pull request #2 from rdovgan/implementation/converter

Added test class for IsConverterService

Havryliuk-Vasyl committed:
Created ConverterService

Implemented methods from IsConverterService

Havryliuk-Vasyl committed:
Added try/catch constructions

Havryliuk-Vasyl committed:
Removed unused import

rdovgan reviewed (changes requested):
@Havryliuk-Vasyl please use formatter for your changes

rdovgan reviewed (changes requested):
The formatter `DateTimeFormatter.ofPattern("yyyy-MM-dd")` is created every time the method convertStringIntoLocalDate is called. It would be more efficient to create a constant that can be reused.

rdovgan reviewed (commented):
In methods like `convertStringIntoInteger`, `convertStringIntoDouble`, and `convertStringIntoLocalDate`, there's a check for null, which is good practice. However, it could be improved by using Optional to better manage the absence of values.

rdovgan reviewed (changes requested):
Using custom messages in `NumberFormatException` and `DateTimeParseException` might create confusion, as calling `e.getMessage()` won't retrieve the original cause of the exception. You could add the original exception as a cause to help with debugging.

Havryliuk-Vasyl committed:
Added Optional to better manage the absence of values

Added Optional to better manage the absence of values, moved DateTimeFormatter and improved throwing exceptions.
